rfcs/45-stageless.md

```diff
@@ -0,0 +1,977 @@
+    // take ownership of the schedule
+    let mut systems = world.resource_mut::<Systems>();
+    let mut schedule = systems.checkout_schedule(MySystemSetLabel);
+
+    // run it on the world
+    schedule.run(world);
+    
+    // return it
+    let mut systems = world.resource_mut::<Systems>();
+    systems.checkin_schedule(schedule);
+}
+```
+
+This pattern is your go-to when your scheduling needs grow beyond "each system runs once per app update", i.e. when you want to:
+
+- repeatedly loop over a sequence of game logic several times in a single app update (e.g. fixed timestep, action queue)
+- have complex branches in your schedule (e.g. state transitions)
+- change to a different threading model for some portion of your systems
+- integrate an external service with arbitrary side effects, like scripting or a web server
+
+As long as you have `&mut World`, from an exclusive system or command, you can extract anything available in the `Systems` resource and run it.
+
+Unlike in previous versions of Bevy, states and the fixed timestep no longer depend on run criteria.
+Instead, systems are simply added under a separate system set that will be retrieved and ran by an exclusive system.
+No conflicts. You can transition states inside the fixed timestep without issue.
+
+### Fixed timestep
+
+A **fixed timestep** advances a fixed number of times each second.
+
+It's implemented as an exclusive system that runs a schedule zero or more times in a row (depending on how long the previous frame took to complete).
+When you supply a constant delta time value (the literal fixed timestep) inside the encapsulated systems, the result is consistent and repeatable behavior regardless of framerate (or even the presence of a GPU).
+
+### States
+
+**State machines** allow for ad-hoc execution of complex logic in response to state transitions.
+They're often used for relatively high-level facets of app behavior like pausing, opening menus, and loading assets.
+State machines are stored as individual resources and their state types must implement the `State` trait.
+It's common to see enums used for state types.
+
+The current state can be read from the `CurrentState<S: State>` resource.
+
+Bevy provides a simple but convenient abstraction to link the transitions of a state `X` with system sets, specifically an `OnEnter(X)` system set and an `OnExit(X)` system set. An exclusive system called `apply_state_transition<S: State>` can be scheduled, which will retrieve and run these schedules as appropriate when a transition is queued in the `NextState<S: State>` resource.
+
+```rust
+fn main() {
+    App::new()
+        /* ... */
+        .add_system(load_map.in_set(OnEnter(GameState::Playing)))
+        .add_system(autosave.in_set(OnExit(GameState::Playing)))
+        /* ... */
+        .run();
+}
+```
+
+## Implementation strategy
+
+Prototype implementation: bevyengine/bevy#4391 (See **Appendix** for specific snippets)

```

Comment:
This PR is now closed and I don't think it adds something over what the RFC presents, since the PR is incomplete and with many compilation errors unsolved last time I checked.

Reply:
```suggestion
```

Reply:
The PR is actually a fairly accurate representation of the proposed changes. The latest error was something from a merge a conflict. I closed it cart's request. We may re-open after this RFC is merged.